includes/polling/os/timos.inc.php

```diff
@@ -10,13 +10,16 @@
 
 $version = 'v' . $majorVersion . '.' . $minorVersion . '.' . $versionModifier;
 
-//SNMPv2-MIB::sysDescr.0 = STRING: TiMOS-B-9.0.R3 both/hops Nokia SAS-Sx 48Tp4SFP+ (PoE) 7210 Copyright (c) 2000-2017 Nokia.
-//All rights reserved. All use subject to applicable license agreements.
-//Built on Thu Jan 5 11:01:16 IST 2017 by builder in /home/builder/9.0B1/R3/panos/main
+$chassis_type_name_array = snmpwalk_cache_oid($device, 'tmnxChassisTypeName', $a = array(), 'TIMETRA-CHASSIS-MIB', 'aos', '-OQUs');
+$hardware = end(reset($chassis_type_name_array));
 
-$pattern = "~(cpm|both)\/(hops64|hops|x86_64) (?'hardware'.*)\sCopyright~";
-preg_match($pattern, $poll_device['sysDescr'], $matches);
-
-if ($matches['hardware']) {
-    $hardware = $matches['hardware'];
+$props = snmpwalk_cache_numerical_oid($device, 'tmnxHwEntry.7', $props = array(), 'TIMETRA-CHASSIS-MIB', 'aos', '-OQne');
```

Comment:
Relatively small, but will vary depending on the device.  On a SAS-K:

```
snmpwalk -M /usr/share/snmp/mibs/:Code/public_projects/librenms/mibs:Code/SNMP-mibs/nokia -m +TIMETRA-CHASSIS-MIB -v2c -c public -OQun 10.1.86.31 tmnxHwEntry.7
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.50331649 = physChassis
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.83886081 = powerSupply
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.134217729 = ioModule
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.150994977 = cpmModule
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.167772162 = fabricModule
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.184549633 = mdaModule
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.201327105 = flashDiskModule
.1.3.6.1.4.1.6527.3.1.2.2.1.8.1.7.1.201327106 = flashDiskModule
```
Again, I can find no way to otherwise identify those IDs and I don't yet have a broad enough pool of devices to test against to see if there is any significant consistency to them.  So walking them and looking for the one identified as "3" (PhysicalChassis/Chassis, depends on the MIB version which is returned) was the best solution I could concoct.